### ISSUE TITLE
.findprovs endpoint is inconsistent

### ISSUE THREAD

**Issue by daviddias** [ipfs/go-ipfs#5047]
#### Description:

.findprovs endpoints doesn't follow the same structure to report errors. Full report here https://github.com/ipfs/js-ipfs-api/pull/782

**Comment by magik6k** [comment]
It looks like it's coming from cmds lib - https://github.com/ipfs/go-ipfs-cmds/blob/master/cli/parse.go#L302 - dit.findprovs says it requires the `key` argument (first bool in https://github.com/ipfs/go-ipfs/blob/master/core/commands/dht.go#L148).

I've looked at the tests (in https://github.com/ipfs/interface-ipfs-core/pull/288/files) and I don't see any case of the arg not being passed, so there likely something weird going on. @diasdavid can you try to capture the requests with tcpdump/wireshark? Seeing them might help.

**Comment by diasdavid** [comment]
@magik6k my issue was specifically on the fact that the error is incorrectly formatted in comparison to all the other endpoints.

**Comment by Mr0grog** [comment]
FWIW, I think the problem is this particular error, not this endpoint. (Maybe we [aren’t picking a response encoder until after this part of the request parsing](https://github.com/ipfs/go-ipfs-cmds/blob/6d7f4fc86b88d912f4f836847b819a50d75c11c6/command.go#L86-L105)?) I get the same plain text response when omitting `?arg=xyz` on a lot of other endpoints, too.

**Comment by Stebalien** [comment]
We need to set the encoding before checking the arguments.

**Comment by Stebalien** [comment]
Closing as won't fix. The default encoder is "text" and we need to parse the arguments before we can figure out that we actually want to send back json. However, we hit this error when parsing the arguments.
